### PR TITLE
Upgrade to tracing-subscriber 0.3

### PR DESCRIPTION
**Pull request by bertof**
Updated the library to use the newer tracing-subscriber traits.
Changed the least amount of code possible and tested checking for regression.
Bumped the library version as the interfaces have changed.

### CODE REVIEW

**Comment by hlb8122** [comment]
Thanks very much. Aiming for a release sometime this week. Will make sure to include this.

**Comment by bertof** [comment]
You're welcome! Thank you for this library, it's really easy to use.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "tracing-gelf"
-version = "0.5.1"
+version = "0.6.0"
```
Comment (hlb8122) [inline Cargo.toml]: Do you mind reverting this? I'd like to bump to `0.6.0` in a~"prepare for release" commit. I released a patch version recently and this is causing merge conflict.

File: src/lib.rs [hunk 2]
```diff
@@ -420,21 +420,27 @@ where
...
-                    object.extend(span_object.clone());
-                }
-                if spans != String::new() {
-                    spans = format!("{}:{}", spans, span.name());
-                } else {
-                    spans = span.name().to_string();
-                }
-
-                spans
+            let span = ctx.current_span().id().and_then(|id| {
```
Comment (hlb8122) [inline src/lib.rs:524]: I did a little jiggle around before `0.6`. Do you mind rebasing against lastest `master`, this should work

```rust
            let span = ctx.current_span().id().and_then(|id| {
                ctx.span_scope(id).map(|scope| {
                    scope.from_root().fold(String::new(), |mut spans, span| {
                        // Add span fields to the base object
                        if let Some(span_object) = span.extensions().get::<HashMap<Cow<'static, str>, Value>>() {
                            object.extend(span_object.clone());
                        }
                        if spans != String::new() {
                            spans = format!("{}:{}", spans, span.name());
                        } else {
                            spans = span.name().to_string();
                        }

                        spans
                    })
                })
            });

            if let Some(span) = span {
                object.insert("_span".into(), span.into());
            }
```

**Review by hlb8122 — CHANGES_REQUESTED** [review]

**Comment by bertof** [comment]
I've updated my PR.

**Comment by hlb8122** [comment]
@bertof Thanks

**Comment by hlb8122** [comment]
@bertof Released in `0.6` 👍

**Pull request merged**